packages/swingset-liveslots/src/virtualObjectManager.js

```diff
@@ -15,137 +15,275 @@ const { details: X, quote: q } = assert;
-            store(entry.baseRef, entry.rawState);
-            entry.dirty = false;
-          }
-          entry = entry.prev;
-        }
-        dirtyCount = 0;
-      }
-    },
-    remember(innerObj) {
-      if (liveTable.has(innerObj.baseRef)) {
-        return;
-      }
-      cache.makeRoom();
-      liveTable.set(innerObj.baseRef, innerObj);
-      innerObj.prev = undefined;
-      innerObj.next = lruHead;
-      if (lruHead) {
-        lruHead.prev = innerObj;
-      }
-      lruHead = innerObj;
-      if (!lruTail) {
-        lruTail = innerObj;
-      }
-      // kdebug(`### vo LRU remember ${lruHead.baseRef}`);
-    },
-    refresh(innerObj) {
-      if (innerObj !== lruHead) {
-        const oldPrev = innerObj.prev;
-        const oldNext = innerObj.next;
-        if (oldPrev) {
-          oldPrev.next = oldNext;
-        } else {
-          lruHead = oldNext;
-        }
-        if (oldNext) {
-          oldNext.prev = oldPrev;
-        } else {
-          lruTail = oldPrev;
-        }
-        innerObj.prev = undefined;
-        innerObj.next = lruHead;
-        lruHead.prev = innerObj;
-        lruHead = innerObj;
-        // kdebug(`### vo LRU refresh ${lruHead.baseRef}`);
-      }
-    },
-    lookup(baseRef, load) {
-      let innerObj = liveTable.get(baseRef);
-      if (innerObj) {
-        cache.refresh(innerObj);
-      } else {
-        innerObj = { baseRef, rawState: null, repCount: 0 };
-        cache.remember(innerObj);
-      }
-      if (load && !innerObj.rawState) {
-        innerObj.rawState = fetch(baseRef);
+function assessFacetiousness(obj, inner) {
+  if (typeof obj !== 'object') {
+    return 'not';
+  }
+  let established;
+  for (const prop of Reflect.ownKeys(obj)) {
+    const value = obj[prop];
+    let current;
+    if (typeof value === 'function') {
+      current = 'one';
+    } else if (
+      !inner &&
+      typeof value === 'object' &&
+      assessFacetiousness(value, true) === 'one'
+    ) {
+      if (typeof prop === 'symbol') {
+        // can't have symbol-named facets
+        return 'not';
       }
-      return innerObj;
-    },
+      current = 'many';
+    } else {
+      return 'not';
+    }
+    if (!established) {
+      established = current;
+    } else if (established !== current) {
+      return 'not';
+    }
+  }
+  if (!established) {
+    // empty objects are methodless Far objects
+    return 'one';
+  } else {
+    return /** @type {Facetiousness} */ (established);
+  }
+}
+
+// To manage context and state and data correctly (not sensitive to GC), we need
+// two Caches. The first is "dataCache", and maps baseRef to state data: {
+// capdata, valueMap }.  "capdata" is a record like { propX: capdataX, propY:
+// capdataY }, one property per property name of the "state" object, initialized
+// from 'initialData' when the instance was first created. "valueMap" is a Map,
+// from "propX" to the unserialized value of `state.propX`. The second cache is
```

Comment:
I think an independent paragraph for each cache would be valuable.
```suggestion
// from "propX" to the unserialized value of `state.propX`.
//
// The second cache is
```
**EDIT**: Disregard if this details are removed from the summary as suggested.